src/tests/Microsoft.PowerFx.Interpreter.Tests/DeferredTypeTests.cs

```diff
@@ -0,0 +1,122 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using Microsoft.PowerFx.Core.Localization;
+using Microsoft.PowerFx.Types;
+using Xunit;
+
+namespace Microsoft.PowerFx.Interpreter
+{
+    public class DeferredTypeTests
+    {
+        [Theory]
+        [InlineData("X")]
+        [InlineData("$\"Test {X}\"")]
+        [InlineData("!X")]
+        [InlineData("Table(X)")]
+        [InlineData("X.Field1")]
+        [InlineData("{test: X}")]
+        [InlineData("RX.XF")]
+        [InlineData("RX.XF + 1")]
+
+        [InlineData("X + 1")]
+        [InlineData("X + \"1\"")]
+        [InlineData("X + DateTime(2022, 11, 10, 0, 0, 0)")]
+        [InlineData("X + Date(2022, 11, 10)")]
+        [InlineData("X + Time(0, 0, 0)")]
+        [InlineData("1 + X")]
+        [InlineData("DateTime(2022, 11, 10, 0, 0, 0) + X")]
+        [InlineData("Date(2022, 11, 10) + X")]
+        [InlineData("Time(0, 0, 0) + X")]
+
+        [InlineData("X * 1")]
+        [InlineData("And(X, 1=1)")]
+        [InlineData("X&\"test\"")]
+        [InlineData("X = 1")]
+        [InlineData("X = GUID(\"5cc45615-f759-4a53-b225-d3a2497f60ad\")")]
+
+        [InlineData("X < 2")]
+        [InlineData("\"The\" in X")]
+        [InlineData("123 in X")] // check
+        [InlineData("X in [1, 2, 3]")] // check
+        [InlineData("123 in Table(X)")]
+        [InlineData("IsBlank(X)")]
+        [InlineData("IsError(X)")]
+        [InlineData("Text(X)")]
+        [InlineData("Value(X)")]
+        [InlineData("Boolean(X)")]
+        [InlineData("Index([1,2,3].Value, X)")]
+
+        // Ensures expression binds without any errors - but issues a warning for the deferred(unknown) type.
+        public void DeferredTypeTest(string script)
+        {
+            Preview.FeatureFlags.StringInterpolation = true;
+            var symbolTable = new SymbolTable();
+
+            // Record with deferred field
+            var rX = RecordType.Empty().Add("XF", FormulaType.Deferred);
+
+            symbolTable.AddVariable("X", FormulaType.Deferred);
+            symbolTable.AddVariable("RX", rX);
+
+            TestDeferredTypeBindingWarning(script, Features.None, symbolTable);
+        }
+
+        [Theory]
+        [InlineData("$\"Test {X} {R}\"", "Invalid argument type (Record). Expecting a Text value instead.")]
+        [InlineData("X + R", "Invalid argument type. Expecting one of the following: Number, Text, Boolean.")]
+        [InlineData("Table(X, N)", "Cannot use a non-record value in this context")]
+        [InlineData("X.field + N.field", "Invalid use of '.'")]
+        [InlineData("Index([1,2,3], X).missing", "Name isn't valid. 'missing' isn't recognized")]
+
```

Comment:
Can we consider adding these tests as well? (just to cover some of our common use cases)

`"X < \"2021-12-09T20:28:52Z\""`
`"true And X.field Or "string""`

Reply:
`"true And X.field Or \"string\""` is considered syntactically valid expression since we coerce string param to Boolean.
e.g. `false Or "true"` => `true`
Added test for other one.